### PR TITLE
Fixes sitemap and proxy calls

### PR DESCRIPTION
**Pull request by bigbasec**
This is a change to the way basic ui looks up sitemap names for Image and Video types(and any other future types that utilize the proxy).

PR for the core.proxy service is here : openhab/openhab-core#2162

Description of issues covered by this :
https://community.openhab.org/t/images-video-sitemaps-and-proxy/115677

### CODE REVIEW

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public boolean canRender(Widget w) {
         snippet = StringUtils.replace(snippet, "%proxied_url%", proxiedUrl);
         snippet = StringUtils.replace(snippet, "%ignore_refresh%", ignoreRefresh ? "true" : "false");
         snippet = StringUtils.replace(snippet, "%url%", url);
-
+        // logger.warn(snippet);
```
Comment (lolodomo) [inline bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java:101]: To be suppressed with logger variable definition.
Other option: use a unvommented TRACE log but add something to your log like "snippet for image widget : {}"

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/VideoRenderer.java [hunk 2]
```diff
@@ -56,12 +56,12 @@ public boolean canRender(Widget w) {
...
 
     @Override
-    public EList<Widget> renderWidget(Widget w, StringBuilder sb) throws RenderException {
+    public EList<Widget> renderWidget(Widget w, StringBuilder sb, String sitemap) throws RenderException {
         Video videoWidget = (Video) w;
         String snippet = null;
 
         String widgetId = itemUIRegistry.getWidgetId(w);
-        String sitemap = w.eResource().getURI().path();
+        // String sitemap = w.eResource().getURI().path();
```
Comment (lolodomo) [inline bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/VideoRenderer.java:64]: No need to keep bad code commented.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java [hunk 3]
```diff
@@ -46,6 +48,8 @@
 
     private static final String URL_NONE_ICON = "images/none.png";
 
+    private final Logger logger = LoggerFactory.getLogger(ImageRenderer.class);
```
Comment (Hilbrand) [inline bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java:51]: Is this logger used?

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java (hunk 1 above)
Comment (bigbasec) [reply bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java]: Removed.

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/VideoRenderer.java (hunk 2 above)
Comment (bigbasec) [reply bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/VideoRenderer.java]: Removed.

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/TextRenderer.java [hunk 4]
```diff
@@ -39,6 +41,8 @@
 @NonNullByDefault
 public class TextRenderer extends AbstractWidgetRenderer {
 
+    private final Logger logger = LoggerFactory.getLogger(TextRenderer.class);
```
[the remark this answers is not in the document]
Comment (bigbasec) [reply bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/TextRenderer.java]: Nope, it's been removed.

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java (hunk 3 above)
Comment (bigbasec) [reply bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java]: It is not, it's been removed.

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java [hunk 5]
```diff
@@ -32,6 +32,8 @@
 import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
```
Comment (lolodomo) [inline bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java:35]: No need for these imports if nothing is logged by this class.

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/TextRenderer.java [hunk 6]
```diff
@@ -27,6 +27,8 @@
 import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
```
Comment (lolodomo) [inline bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/TextRenderer.java:30]: No need for these imports if nothing is logged by this class.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/TextRenderer.java (hunk 6 above)
Comment (bigbasec) [reply bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/TextRenderer.java]: Good catch, thanks!

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java (hunk 5 above)
Comment (bigbasec) [reply bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/ImageRenderer.java]: Removed, thank you!

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by bigbasec** [comment]
> LGTM

Just a heads up here, this won't fix the proxy part of things, it just fixes the rendering for the widgets.   

https://github.com/openhab/openhab-core/pull/2162 would resolve the proxy side so things actually work.

**Comment by ghys** [comment]
@bigbasec are there dependencies between these PR i.e. should we wait for the others to be merged first?

**Comment by bigbasec** [comment]
> @bigbasec are there dependencies between these PR i.e. should we wait for the others to be merged first?

This won't have any effect on a system that doesn't have the core PR, no.   The only 2 widgets that use the "sitemap" variable are Image and Video, and those are also the only 2 that need the proxy.   

So no, putting this in won't affect anything, but alone it won't fix the images and videos.

**Comment by bigbasec** [comment]
> @bigbasec are there dependencies between these PR i.e. should we wait for the others to be merged first?

Dependencies have been merged.

**Comment by bigbasec** [comment]
Getting the hang of this git thing.  That should resolve the build issue.

**Comment by bigbasec** [comment]
Anything I need to do here to get this merged in?   Although there wasn't any dependency on this from the core, this is required to get the core working again.   Explained here : https://github.com/openhab/openhab-core/pull/2178

**Review by kaikreuzer — APPROVED** [review]
All fine, sorry for the delay and many thanks!

**Pull request merged**